src/test/java/InteractionGroupServiceTest.java

```diff
@@ -40,17 +40,15 @@ public void groupByClientValidInputTest() {
         testValue.add(interaction2);
         testValue.add(interaction3);
 
-        HashMap<String, List<Interaction>> hashMap = new HashMap<>();
-        groupService.groupByClient(testValue, hashMap);
+        HashMap<String, List<Interaction>> hashMap = new HashMap<>(groupService.groupByClient(testValue));
```

Comment:
"[Yo dawg](https://www.urbandictionary.com/define.php?term=Yo%20Dawg), so I've heard that you love HashMaps, so we put HasMap in your HasMap so you can..."
You don't need this extra layer here, just use the value returned by your method.